### PR TITLE
SVCS-2025 Add findAll and findAllIterator

### PR DESCRIPTION
**Pull request by AaronTrippaers**

### CODE REVIEW

File: src/services/dispatchers/dispatchers/types.ts [hunk 1]
```diff
@@ -19,6 +20,30 @@ export interface DispatchersService {
    */
   find(options?: OptionsWithRql): Promise<PagedResult<Dispatcher>>;
 
+  /**
```
Comment (AaronTrippaers) [inline src/services/dispatchers/dispatchers/types.ts:23]: Double check the jsdocs 😄  Did not find good examples for these 😬

File: src/services/dispatchers/dispatchers/index.ts [hunk 2]
```diff
@@ -40,6 +41,14 @@ export default (
       return await query(options);
     },
 
+    async findAll(this: DispatchersService, options) {
+      return findAllGeneric<Dispatcher>(this.find, options);
```
Comment (chad-pelly) [inline src/services/dispatchers/dispatchers/index.ts:45]: I think both instances of `this.find` should be `query` 🤔

File: src/services/dispatchers/dispatchers/types.ts (hunk 1 above)
Comment (chad-pelly) [reply src/services/dispatchers/dispatchers/types.ts:23]: Regarding the comments on the interfaces for the Dispatcher findAll / findAll iterator:
https://github.com/ExtraHorizon/javascript-sdk/blob/358ec06cb9aad18774d7dfac1f607c6d399fa948/src/services/auth/oidc/providers/types.ts

**Review by chad-pelly — COMMENTED** [review]
Just two very small comments

File: src/services/dispatchers/dispatchers/index.ts (hunk 2 above)
Comment (AaronTrippaers) [reply src/services/dispatchers/dispatchers/index.ts:45]: Copied from: https://github.com/ExtraHorizon/javascript-sdk/blob/358ec06cb9aad18774d7dfac1f607c6d399fa948/src/services/tasks/tasks.ts but there we don't have query, so I will change it 😄

File: src/services/dispatchers/dispatchers/types.ts [hunk 3]
```diff
@@ -19,6 +20,31 @@ export interface DispatchersService {
...
+  findAll(options?: OptionsWithRql): Promise<Dispatcher[]>;
+
+  /**
+   * ## Retrieve a paged list of Dispatchers
+   *
+   * **Global Permissions:**
+   * - `VIEW_DISPATCHERS` - Allows a user to view Dispatchers
+   *
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns An iterator for the queried providers {@link FindAllIterator FindAllIterator<Dispatcher>}
```
Comment (chad-pelly) [inline src/services/dispatchers/dispatchers/types.ts:43]: `providers` -> `Dispatchers` 😜

File: src/services/dispatchers/dispatchers/types.ts [hunk 4]
```diff
@@ -19,6 +20,31 @@ export interface DispatchersService {
...
+   * @throws {@link NoPermissionError} when the user doesn't have the required permissions to execute the function.
```
Comment (chad-pelly) [inline src/services/dispatchers/dispatchers/types.ts:44]: This is also not related to functions and we decided to not add the errors here I think 🤔

Comment (chad-pelly) [reply src/services/dispatchers/dispatchers/types.ts:44]: Same comments in the findAll 😂

**Review by chad-pelly — COMMENTED** [review]
Few minor things still 😂

**Review by chad-pelly — APPROVED** [review]
